### PR TITLE
- Added WEB supporting

### PR DESCRIPTION
**Pull request by NikolayFatkullin**

### CODE REVIEW

File: src/main/java/spring/intro/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        userService.add(thor);
+        return "All users are successfully added to the DB";
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable(name = "userId") Long userId) {
+        return userMapper.mapToDto(userService.getById(userId).get());
+    }
+
+    @GetMapping("/")
```
Comment (Fedich09) [inline src/main/java/spring/intro/controller/UserController.java:51]:
```suggestion
    @GetMapping
```

**Review by Fedich09 — APPROVED** [review]

File: src/main/java/spring/intro/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -46,4 +47,13 @@ public void add(User user) {
             throw new DataProcessingException("Can't get all users", e);
         }
     }
+
+    @Override
+    public Optional<User> getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.of(session.get(User.class, id));
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/dao/impl/UserDaoImpl.java:54]: there is no point in using Optional in such way. What will happen if the null will be passed to the method of()?

File: src/main/java/spring/intro/dto/UserResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package spring.intro.dto;
+
+public class UserResponseDto {
+    private String name;
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/dto/UserResponseDto.java:4]: let's add id too

File: src/main/java/spring/intro/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -23,4 +24,9 @@ public void add(User user) {
     public List<User> listUsers() {
         return userDao.listUsers();
     }
+
+    @Override
+    public Optional<User> getById(Long id) {
+        return userDao.getById(id);
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/service/impl/UserServiceImpl.java:31]: service should return an object or throw an exception(you can use orElseThrow for this) but not Optional

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**